File: boj/brute-force/2529.swift
import Foundation

let range = 10

var k = Int()
var marks = [String]()
var visited = [Bool]()
var result = [String]()

var maxRes = "", minRes = "\(Int.max)"

func search(_ num: Int, _ order: Int, _ index: Int) {
    if order == marks.count {
        let resNum = result.joined(separator: "")
        maxRes = max(maxRes, resNum)
        minRes = min(minRes, resNum)
        return
    }
    
    for nextNum in 0..<range {
        if visited[nextNum] { continue }
        
        visited[nextNum] = true
        if marks[order] == "<" {
            if num < nextNum {
                result[index] = "\(nextNum)"
                search(nextNum, order + 1, index + 1)
            }
        }
        else {
            if num > nextNum {
                result[index] = "\(nextNum)"
                search(nextNum, order + 1, index + 1)
            }
        }
        visited[nextNum] = false
    }
    
}

func solution() {
    k = Int(readLine()!)!
    marks = readLine()!.split(separator: " ").compactMap { String($0) }
    let range = 10
    
    result = [String](repeating: String(), count: k + 1)
    visited = [Bool](repeating: false, count: range)
    
    for num in 0..<range {
        visited[num] = true
        result[0] = "\(num)"
        search(num, 0, 1)
        visited[num] = false
    }
    
    print(maxRes)
    print(minRes)
}

solution()

File: boj/brute-force/2529.java
import javax.jws.Oneway;
import java.util.*;
import java.io.*;

public class Main {

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    static int k;
    static StringBuffer max = new StringBuffer("-1"), min = new StringBuffer("10000000000");
    static int[] result;
    static String[] oper;
    static boolean[] visited ;
    static final int RANGE = 10;

    public static void main(String[] args) throws IOException {
        k = Integer.parseInt(br.readLine());
        oper = br.readLine().split(" ");

        result = new int[oper.length + 1];
        visited = new boolean[RANGE];

        for (int i = 0; i < RANGE; i++) {
            visited[i] = true;
            result[0] = i;
            search(i, 1, 0);
            visited[i] = false;
        }

        bw.write(max.toString() + "\n" + min.toString());
        bw.flush();
    }

    static void search(int number, int index, int operIdx) {
        if (operIdx == oper.length) {
            conversion();
        }
        else {
            for (int i = 0; i < RANGE; i++) {
                if (visited[i]) continue;

                visited[i] = true;
                if (oper[operIdx].equals("<")) {
                    if (number < i) {
                        result[index] = i;
                        search(i, index + 1, operIdx + 1);
                    }
                }
                if (oper[operIdx].equals(">")) {
                    if (number > i) {
                        result[index] = i;
                        search(i, index + 1, operIdx + 1);
                    }
                }

                visited[i] = false;
            }
        }
    }

    static void conversion() {
        StringBuffer sum = new StringBuffer();
        for (int ele : result) {
            sum.append(ele);
        }

        if (Long.parseLong(max.toString()) < Long.parseLong(sum.toString()))
            max = sum;
        if (Long.parseLong(min.toString()) > Long.parseLong(sum.toString()))
            min = sum;
    }

}